webhook/server/server.go

```diff
@@ -0,0 +1,437 @@
+	if err != nil {
+		err := logging.Errorf("could not get Network Attachment Defintion %s/%s: %s", namespace, name, err)
+		return nil, err
+	}
+
+	networkAttachmentDefinition := types.NetworkAttachmentDefinition{}
+	json.Unmarshal(rawNetworkAttachmentDefinition, &networkAttachmentDefinition)
+
+	return &networkAttachmentDefinition, nil
+}
+
+func parsePodNetworkSelectionElement(selection string) (*types.NetworkSelectionElement, error) {
+	var namespace, name, netInterface string
+	var networkSelectionElement *types.NetworkSelectionElement
+
+	units := strings.Split(selection, "/")
+	switch len(units) {
+	case 1:
+		namespace = "default"
+		name = units[0]
+	case 2:
+		namespace = units[0]
+		name = units[1]
+	default:
+		return networkSelectionElement, logging.Errorf("invalid network selection element - more than one '/' rune in: %s", selection)
+	}
+
+	units = strings.Split(name, "@")
+	switch len(units) {
+	case 1:
+		name = units[0]
+		netInterface = ""
+	case 2:
+		name = units[0]
+		netInterface = units[1]
+	default:
+		return networkSelectionElement, logging.Errorf("invalid network selection element - more than one '@' rune in: %s", selection)
+	}
+
+	validNameRegex, _ := regexp.Compile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`)
+	for _, unit := range []string{namespace, name, netInterface} {
+		ok := validNameRegex.MatchString(unit)
+		if !ok && len(unit) > 0 {
+			return networkSelectionElement, logging.Errorf("at least one of the network selection units is invalid: error found at '%s'", unit)
+		}
+	}
+
+	networkSelectionElement = &types.NetworkSelectionElement{
+		Namespace:        namespace,
+		Name:             name,
+		InterfaceRequest: netInterface,
+	}
+
+	return networkSelectionElement, nil
+}
+
+func parsePodNetworkSelections(podNetworks string) ([]*types.NetworkSelectionElement, error) {
+	var networkSelections []*types.NetworkSelectionElement
+
+	if len(podNetworks) == 0 {
+		return nil, logging.Errorf("empty string passed as network selection elements list")
+	}
+
+	/* try to parse as JSON array */
+	err := json.Unmarshal([]byte(podNetworks), &networkSelections)
+
+	/* if failed, try to parse as comma separated */
+	if err != nil {
+		logging.Debugf("%s is not in JSON format: %s. Will try to parse as comma separated network selections list", podNetworks, err.Error())
+		for _, networkSelection := range strings.Split(podNetworks, ",") {
+			networkSelection = strings.TrimSpace(networkSelection)
+			networkSelectionElement, err := parsePodNetworkSelectionElement(networkSelection)
+			if err != nil {
+				return nil, logging.Errorf("error parsing network selection element: %v", err)
+			}
+			networkSelections = append(networkSelections, networkSelectionElement)
+		}
+	}
+
+	/* fill missing namespaces with default value */
+	for _, networkSelection := range networkSelections {
+		if networkSelection.Namespace == "" {
+			networkSelection.Namespace = "default"
```

Comment:
It should be the pod's namespace not necessarily "default".

Reply:
done